Use match_all instead of asterisk for blank search

trevan opened the pull request:
In https://github.com/elastic/kibana/pull/10280#issuecomment-279593482, @lukasolson said that Kibana should use the match_all query instead of the asterisk for default searches.  These are the changes to accomplish that.

elasticmachine commented:
Can one of the admins verify this patch?

lukasolson commented on src/ui/public/parse_query/lib/to_user.js:
```diff
@@ -10,6 +10,7 @@ export default function toUser(text) {
+    if (text.match_all && !_.has(text.match_all, 'boost')) return '';
     return angular.toJson(text);
   }
   if (text === '*') {
```
If the default is now match_all, is this part necessary? I understand there are saved searches that already have "\*" as the query, but with your changes, if you delete the "\*" then you'll have an empty query (so the placeholder can show) and the default will no longer be "\*". I think it would be more intuitive if we ripped this out with these changes.

lukasolson replied:
(I'm referring to [these lines](https://github.com/trevan/kibana/blob/1922565f6cf0117973168131bc661c8755cb9d97/src/ui/public/parse_query/lib/to_user.js#L16-L18).)

trevan replied:
Good point.  I removed the lines.  Though the '*' is in not just saved searches, but saved dashboards and visualizations as well.  So pretty much all saved items will show '*' unless the user has added a specific query.

lukasolson commented:
This is definitely what I had in mind. Thanks for the PR, @trevan.

trevan commented:
@lukasolson, I was wondering where the tests were.  I've fixed them but I'm wondering if they should be moved to the parse_query directory or if the parse_query directive should be moved to the directives directory.
I also updated visualize and dashboard.

lukasolson commented:
jenkins, test this

Bargs commented:
I couldn't quite put my finger on it before, but I'm concerned about spreading knowledge of the `match_all` query syntax everywhere. We shouldn't have to do stuff like [this](https://github.com/trevan/kibana/blob/b04f75a26006739ca7d70627b2cf33dd68887552/src/ui/public/parse_query/lib/to_user.js#L13), but we do because we're now blurring the lines between query string queries and Query DSL queries. We could do this in a much cleaner manner once we've established our query model. What do you say to putting this PR on hold until then @lukasolson? Unless there's a big performance win here, I think we might just be creating more complexity.

lukasolson commented:
@Bargs I'm fine with putting this on hold for now. I mentioned this PR to @spalger and he had a good point that this does introduce a little bit of inconsistency, and I tend to agree. Seems like there needs to be a bit more conversation around this before it's merged.

Bargs commented:
Gonna close this for now since we'll revisit later.
